src/content/drafts/how-to-create-a-custom-eslint-rule.md

```diff
@@ -0,0 +1,286 @@
+
+By thinking through this *now*, you can not only make sure you actually want to implement this rule, but you can also make sure that you truly know what you're getting yourself into. Furthermore, this will also make writing tests trivial.
+
+For the bacon example, let's go with the plan that you'll add a warning for all identifiers that contain `bacon` while also ignoring cases (so `somebAcon` and `baConstructor` would both be rejected). However, it wouldn't go as far as adding searches for underscores or leet speak (so `ba_con` and `b4c0n` would both be accepted).
+
+Here are some examples that would cause this rule to **fail** (meaning the user gets a warning):
+
+```javascript
+let bacona = "hello";
+console.log(bacon);
+function baConstructor() {};
+function validName(bacon) {};
+class someBaconClass {};
+```
+
+Here are some examples that would cause this rule to **pass** (meaning no warning):
+
+```javascript
+let pork = 1;
+console.log('bacon');
+function ba_con() {};
+class b4c0n {};
+```
+
+## Who will benefit from this rule?
+
+Now that you have some idea of how your rule will work, you should take some time to think about who would benefit from this rule. Some questions to ask yourself are:
+
+* Is this a rule that would benefit just your company or is it something that other people might want?
+* If this rule **is** specific to your company:
+  * Is it specific to a single project?
+  * Is it something that many different projects could benefit from?
+* If this rule **isn't** specific to your company:
+  * Is it something that benefits *all* JavaScript projects or is it more specific than that? For example, a rule that provides suggestions for TypeScript apps that use GitHub would be something that is beneficial for some projects, whereas a rule that suggests to not use `console.log` would be beneficial for all types of projects. 
+
+Once you have those answers you can then think about...
+
+## Where will the code for the rule live?
+
+There are four main places that new rules are commonly made:
+
+1. In the core [ESLint repository](https://github.com/eslint/eslint/tree/main/lib/rules) 
+    * *For rules that are useful to all JavaScript projects*
+2. In a plugin that already exists 
+    * *For rules that are useful to a specific subset of JavaScript projects*
+3. In a new plugin 
+    * *For rules that are useful to a specific subset of JavaScript projects, but that don't fit into any existing plugin*
+4. Locally in a specific project by utilizing the plugin [eslint-plugin-local-rules](https://www.npmjs.com/package/eslint-plugin-local-rules) 
```

Comment:
You don't actually need to use a plugin to use a local rule. You can use the deprecated `--rulesdir`, but also, you can create a local plugin and reference it in your dependencies using the [`file:` prefix](https://stackoverflow.com/questions/14381898/local-dependency-in-package-json/14387210#14387210).